ui/app/controllers/settings/tokens.js

```diff
@@ -48,31 +49,83 @@ export default class Tokens extends Controller {
     return this.store.peekAll('auth-method');
   }
 
+  get hasJWTAuthMethods() {
+    return this.authMethods.any((method) => method.type === 'JWT');
+  }
+
+  get nonTokenAuthMethods() {
+    return this.authMethods.rejectBy('type', 'JWT');
+  }
+
   @action
-  verifyToken() {
+  async verifyToken() {
     const { secret } = this;
-    this.clearTokenProperties();
     const TokenAdapter = getOwner(this).lookup('adapter:token');
 
-    this.set('token.secret', secret);
-    this.set('secret', null);
+    // const isJWT = secret.length > 36 && secret.split('.').length === 3; // TODO: TEMP, HACKY HACK
+    // use regex to determine if "secret" is a JWT
+    const isJWT =
+      secret.length > 36 &&
+      secret.match(/^[a-zA-Z0-9-_]+\.[a-zA-Z0-9-_]+\.[a-zA-Z0-9-_]+$/);
+
+    if (isJWT) {
+      // TODO: Discussion: There may be multiple JWT methods sometime in the future.
+      // A quick assumption would be to just use the first one, but that
+      // won't hold up forever. We should probably have a way to select which one
+      // to use, or at least a way to know which one was used to generate the token.
+      const methodName = await this.authMethods.findBy('type', 'JWT')?.name;
```

Comment:
Wanted to make sure this was noted in the PR for discussion: If the agent has multiple authMethods of type `JWT`, this will simply pick the first one and assume that's what the user is attempting to sign in with.

It's very easy for this assumption to be wrong — a user could have multiple JWT auth methods and intend on using one other than the first — but it seems like a real edge case of auth method config.

A straightforward solution to this would be to add a "Which JWT method did you want to sign in with?" prompt, or a button set similar to how we handle OIDC auth methods, but then it becomes pretty cumbersome to an end-user.

My current feeling is that we should ship this with a single JWT auth method supported, and worry about interface expansion later if it becomes pressing.

cc @pkazmierczak @juliezzhou @ChaiWithJai

Reply:
I agree that from a UI perspective your current design is neater (kudos btw), but I'm not sure about this: 

> a user could have multiple JWT auth methods and intend on using one other than the first — but it seems like a real edge case of auth method config.

Here's why: JWT is mostly going to be used as a service-to-service/machine-to-machine auth method. Thus it's rather likely a user could have multiple JWT-type auth methods with different role/policy bindings or different methods of validation depending on the use case, or depending where they got that token from. I might be wrong but in contrast to OIDC I'd say it's more likely for people to have multiple JWT methods and a single OIDC method. 

@mikenomitch what do you think about this?